dev/tools/gen_defaults/lib/switch_template.dart

```diff
@@ -0,0 +1,187 @@
+      }
+      if (states.contains(MaterialState.pressed)) {
+        return ${componentColor('md.comp.switch.unselected.pressed.handle')};
+      }
+      return ${componentColor('md.comp.switch.unselected.handle')};
+    });
+  }
+
+  @override
+  MaterialStateProperty<Color> get trackColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        if (states.contains(MaterialState.selected)) {
+          return ${componentColor('md.comp.switch.disabled.selected.track')}.withOpacity(${opacity('md.comp.switch.disabled.track.opacity')});
+        }
+        return ${componentColor('md.comp.switch.disabled.unselected.track')}.withOpacity(${opacity('md.comp.switch.disabled.track.opacity')});
+      }
+      if (states.contains(MaterialState.selected)) {
+        if (states.contains(MaterialState.hovered)) {
+          return ${componentColor('md.comp.switch.selected.hover.track')};
+        }
+        if (states.contains(MaterialState.focused)) {
+          return ${componentColor('md.comp.switch.selected.focus.track')};
+        }
+        if (states.contains(MaterialState.pressed)) {
+          return ${componentColor('md.comp.switch.selected.pressed.track')};
+        }
+        return ${componentColor('md.comp.switch.selected.track')};
+      }
+      if (states.contains(MaterialState.hovered)) {
+        return ${componentColor('md.comp.switch.unselected.hover.track')};
+      }
+      if (states.contains(MaterialState.focused)) {
+        return ${componentColor('md.comp.switch.unselected.focus.track')};
+      }
+      if (states.contains(MaterialState.pressed)) {
+        return ${componentColor('md.comp.switch.unselected.pressed.track')};
+      }
+      return ${componentColor('md.comp.switch.unselected.track')};
+    });
+  }
+
+  @override
+  MaterialStateProperty<Color?> get overlayColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.selected)) {
+        if (states.contains(MaterialState.hovered)) {
+          return ${componentColor('md.comp.switch.selected.hover.state-layer')};
+        }
+        if (states.contains(MaterialState.focused)) {
+          return ${componentColor('md.comp.switch.selected.focus.state-layer')};
+        }
+        if (states.contains(MaterialState.pressed)) {
+          return ${componentColor('md.comp.switch.selected.pressed.state-layer')};
+        }
+        return null;
+      }
+      if (states.contains(MaterialState.hovered)) {
+        return ${componentColor('md.comp.switch.unselected.hover.state-layer')};
+      }
+      if (states.contains(MaterialState.focused)) {
+        return ${componentColor('md.comp.switch.unselected.focus.state-layer')};
+      }
+      if (states.contains(MaterialState.pressed)) {
+        return ${componentColor('md.comp.switch.unselected.pressed.state-layer')};
+      }
+      return null;
```

Comment:
Should this be Colors.transparent? How do we expect overlayColor=null to be interpreted in this case?

CC @darrenaustin - Maybe this is another case where we want null to mean "no overlay" instead of "no preference".

Reply:
When the switch is not in any of the states(hovered, focused, and pressed), should we just give a null value for the state layer?